plugins/kvscheduler/internal/registry/registry_impl.go

```diff
@@ -111,8 +112,11 @@ func (reg *registry) GetDescriptorForKey(key string) *KVDescriptor {
 	var keyDescriptor *KVDescriptor
 	for _, descriptor := range reg.descriptors {
 		if descriptor.KeySelector(key) {
+			if keyDescriptor != nil {
```

Comment:
Was just thinking if this is actually not gonna make `GetDescriptorForKey` more expensive performance-wise. Can you run the perf tests with 20k to compare difference?

Reply:
It varies quite a bit on my system, but there doesn't seem to measurable drop in the performance:
```
BEFORE:

level=info msg="Client #0 done, 20000 tunnels took 1m26.308s" loc="grpc-perf/main.go(302)" logger=grpc-stress-test-client
level=info msg="All clients done, took: 1m16.7757s" loc="grpc-perf/main.go(219)" logger=grpc-stress-test-client
level=info msg="All clients done, took: 1m15.096s" loc="grpc-perf/main.go(219)" logger=grpc-stress-test-client

AFTER:

level=info msg="All clients done, took: 1m5.0734s" loc="grpc-perf/main.go(219)" logger=grpc-stress-test-client
level=info msg="All clients done, took: 1m18.1164s" loc="grpc-perf/main.go(219)" logger=grpc-stress-test-client
level=info msg="All clients done, took: 1m12.1412s" loc="grpc-perf/main.go(219)" logger=grpc-stress-test-client
```

This lookup is cached (for the last 500 different keys), so effectively each distinct key goes through the cycle only once and the number of descriptors isn't that significant.

Reply:
Awesome, merging then.